File: Baekjun/24.9.3/12945. 재미있는 박스 정리/solution.js
/** @format */

const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().split("\n");

let line = 0;
let N = Number(input[line++]);

let [l, r] = [0, Math.floor(N / 2)];
const A = [];
for (let i = 0; i < N; i++) {
  A.push(Number(input[line++]));
}

A.sort();
let ans = 0;

while (r < N && l < Math.floor(N / 2)) {
  if (A[r] / A[l] >= 2) {
    ans += 1;
    l += 1;
  }
  r += 1;
}

console.log(N - ans);

File: Baekjun/24.9.3/12945. 재미있는 박스 정리/solution.py
N = int(input())
A = [int(input()) for _ in range(N)]
A.sort()

l, r = 0, N // 2

ans = 0

while l < N // 2 and r < N:
    if A[r] // A[l] >= 2:
        ans += 1
        l += 1
    r += 1
print(len(A) - ans)
